h1st/model/ensemble/stack_ensemble.py

```diff
@@ -22,28 +17,32 @@ def __init__(self, ensembler, sub_models: List[PredictiveModel], **kwargs):
             **submodel_predict_input_key (str): the default value of submodel_input_key is 'X' \n
             **submodel_predict_output_key (str): the default value of submodel_output_key is 'predictions'
         """
-        super().__init__(ensembler, sub_models)
+        self.ensembler = ensembler
+        self._sub_models = sub_models
         self._submodel_predict_input_key = kwargs.get('submodel_predict_input_key', 'X')
         self._submodel_predict_output_key = kwargs.get('submodel_predict_output_key', 'predictions')
 
     @classmethod
-    def preprocess(cls, sub_models: List[PredictiveModel], submodel_predict_input_key: str, submodel_predict_output_key: str, X: Any) -> Any:
+    def preprocess(cls, sub_models: List[PredictiveModel],
+                        submodel_predict_input_key: str,
+                        submodel_predict_output_key: str,
+                        X: Any) -> Any:
         """
         Predicts all sub models then merge input raw data with predictions for ensembler's training or prediction
         :param X: raw input data
         """
         if isinstance(X, list):
-            predictions = [StackEnsemble.get_submodels_prediction(sub_models,
+            predictions = [np.hstack((x, StackEnsemble.get_submodels_prediction(sub_models,
```

Comment:
@TgithubJ I added code to combine input data with predictions here

Reply:
@nhanitvn Awesome-!